src/utils/withTestMock.tsx

```diff
@@ -0,0 +1,21 @@
+import { ComponentType } from "react";
+
+let useMock = process.env.NODE_ENV === "test";
+
+/**
+ * Toggles our mock DateField/RichTextField components.
+ *
+ * This defaults to true in test suites, but can be disabled if you
+ * _really_ want to test against the real component.
+ */
+export function setUseMockComponents(value: boolean) {
+  useMock = value;
+}
+
+// HOC that conditionally swaps components
+export function withTestMock<P extends object>(Component: ComponentType<P>, MockComponent: ComponentType<P>) {
```

Comment:
The bundling means our "trix doesn't work in jsdom" mocking in `internal-frontend` doesn't work anymore:

```
-jest.mock("@homebound/beam/dist/inputs/RichTextField", () => {
-  return require("@homebound/beam/dist/inputs/RichTextField.mock");
-});
```

So had to move the "swap test component" mocking into beam proper.